charts/apisix/templates/configmap.yaml

```diff
@@ -250,9 +252,24 @@ data:
     {{- end }}
 
     deployment:
-      role: traditional
+      role: {{ .Values.deployment.role }}
+      {{- if or (eq .Values.deployment.role "traditional") (eq .Values.deployment.role "control_plane") }}
+
+      {{- if eq .Values.deployment.role "traditional" }}
       role_traditional:
         config_provider: etcd
+      {{- end }}
+
+      {{- if eq .Values.deployment.role "control_plane" }}
```

Comment:
Else if?

Reply:
It won't make a difference, it's just a personal habit